### PR TITLE
Extended SYSTEM and HAZARD to capture assurance information.

### PR DESCRIPTION
**Pull request by AbhaMoitra**
We are still waiting on deciding what extensions are needed for Lockheed Martin, but this extends SYSTEM and HAZARD core ontology with high-level assurance related classes and properties.

### CODE REVIEW

File: RACK-Ontology/ontology/HAZARD.sadl [hunk 1]
```diff
@@ -12,19 +12,31 @@
...
 
+	// Note that we are thinking of adding a human readable "description" to THING, then we could use that instead of "effect".
+	effect (note "consequence(s) of the HAZARD") describes HAZARD with a single value of type string.
+   
+	severity (note "the severity of the HAZARD") describes HAZARD with a single value of type float.
+
+	likelihood (note "the likelihood of the HAZARD happening") describes HAZARD with a single value of type float.
+
+	// A Hazard can be mitigated by a requirement, operating procedure etc., mitigation by REQUIREMENT is defined in REQUIREMENTS.sadl.
+	H:mitigates (note "a HAZARD may be mitigated by a OP_PROCEDURE") describes OP_PROCEDURE with values of type HAZARD.
```
Comment (glguy) [inline RACK-Ontology/ontology/HAZARD.sadl:38]: This seems like it belongs grouped under the definition of `OP_PROCEDURE`. Is there a particular reason it's on the `HAZARD` side?

File: RACK-Ontology/ontology/SYSTEM.sadl [hunk 2]
```diff
@@ -50,3 +56,27 @@ SYSTEM_DEVELOPMENT
...
+
+FUNCTION  
+	(note "defines functionality / capability")
+	is a type of ENTITY.
+
+	// If we introduce "parent" property for THING, then we could drop parentFunction.	
+	parentFunction (note "allows for decomposition of functionality") describes FUNCTION with a single value of type FUNCTION.
+	parentFunction is a type of wasDerivedFrom.
+	
+	// performedBy is inverse of function; so could be dropped.
```
Comment (glguy) [inline RACK-Ontology/ontology/SYSTEM.sadl:68]: Could we encode directly in the SADL file with

```
performedBy is the inverse of function.
```

?

File: RACK-Ontology/ontology/SYSTEM.sadl [hunk 3]
```diff
@@ -50,3 +56,27 @@ SYSTEM_DEVELOPMENT
...
+	performedBy (note "SYSTEM that is providing the functionality") describes FUNCTION with values of type SYSTEM.
+	performedBy is a type of wasDerivedFrom.
+
+// Later we could add more structure to OP_ENV.
+OP_ENV
```
Comment (glguy) [inline RACK-Ontology/ontology/SYSTEM.sadl:73]: What things do we anticipate would relate to one of these?

File: RACK-Ontology/ontology/HAZARD.sadl (hunk 1 above)
Comment (AbhaMoitra) [reply RACK-Ontology/ontology/HAZARD.sadl:38]: It avoids circularity (at the point of OP_PROCEDURE definition, HAZARD has not been defined).

File: RACK-Ontology/ontology/SYSTEM.sadl (hunk 3 above)
Comment (AbhaMoitra) [reply RACK-Ontology/ontology/SYSTEM.sadl:73]: We will add more details as we understand more about how other performers will structure assurance cases.

**Review by kityansiu — APPROVED** [review]
Abha walked me through the changes. I'm fine with them.

**Pull request merged**